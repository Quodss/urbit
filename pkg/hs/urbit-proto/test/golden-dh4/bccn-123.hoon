=/  one  $1
=/  two  ?(%1 %2)
=/  tre  ?(%1 %2 %3)
=/  ott
  $%  [%one one]
      [%two two]
      [%tre tre]
  ==
%.  [%one 1]
|=  t=ott
^-  @
?-  t
  [_ 1]     1
  [%two 2]  2
  [%tre 2]  3
  [%tre 3]  3
==
