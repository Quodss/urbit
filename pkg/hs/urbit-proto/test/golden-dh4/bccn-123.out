PARSED

=/  one
  $1
=/  two
  ?(%1 %2)
=/  tre
  ?(%1 %2 %3)
=/  ott
  $%  {$one [one]}
      {$two [two]}
      {$tre [tre]}
  ==
%.  [%one 1]
|=  t=ott
^-  @
?-(t [_ 1] 1, [%two 2] 2, [%tre 2] 3, [%tre 3] 3)

SOFT

=+  one=$1
=+  two=?(%1 %2)
=+  tre=?(%1 %2 %3)
=+  ^=  ott
    $:  ?(%6647407 %6648436 %7305076)
        ^-  $
        ?:  ?=(%"one" +2)
          one
        ?:  ?=(%"two" +2)
          two
        ??  ?=(%"tre" +2)
        tre
    ==
%+  |=  t=ott
    ^-  @
    ?:  ?=([_ %1] t)
      1
    ?:  ?=([%"two" %2] t)
      2
    ?:  ?=([%"tre" %2] t)
      3
    ??  ?=([%"tre" %3] t)
    3
  %one
1

ERROR

  trace:  play:
          lvl   0
          sut   0|*
          ----
          code  =+  one=$1
                =+  two=?(%1 %2)
                =+  tre=?(%1 %2 %3)
                =+  ^=  ott
                    $:  ?(%6647407 %6648436 %7305076)
                        ^-  $
                        ?:  ?=(%"one" +2)
                          one
                        ?:  ?=(%"two" +2)
                          two
                        ??  ?=(%"tre" +2)
                        tre
                    ==
                %+  |=  t=ott
                    ^-  @
                    ?:  ?=([_ %1] t)
                      1
                    ?:  ?=([%"two" %2] t)
                      2
                    ?:  ?=([%"tre" %2] t)
                      3
                    ??  ?=([%"tre" %3] t)
                    3
                  %one
                1


          play:
          lvl   1
          sut   :-  $=  one=$1
                    one=$
                0|*

          ----
          code  =+  two=?(%1 %2)
                =+  tre=?(%1 %2 %3)
                =+  ^=  ott
                    $:  ?(%6647407 %6648436 %7305076)
                        ^-  $
                        ?:  ?=(%"one" +2)
                          one
                        ?:  ?=(%"two" +2)
                          two
                        ??  ?=(%"tre" +2)
                        tre
                    ==
                %+  |=  t=ott
                    ^-  @
                    ?:  ?=([_ %1] t)
                      1
                    ?:  ?=([%"two" %2] t)
                      2
                    ?:  ?=([%"tre" %2] t)
                      3
                    ??  ?=([%"tre" %3] t)
                    3
                  %one
                1


          play:
          lvl   2
          sut   :+  $=  two=?(%1 %2)
                    two=$
                  $=  one=$1
                  one=$
                0|*

          ----
          code  =+  tre=?(%1 %2 %3)
                =+  ^=  ott
                    $:  ?(%6647407 %6648436 %7305076)
                        ^-  $
                        ?:  ?=(%"one" +2)
                          one
                        ?:  ?=(%"two" +2)
                          two
                        ??  ?=(%"tre" +2)
                        tre
                    ==
                %+  |=  t=ott
                    ^-  @
                    ?:  ?=([_ %1] t)
                      1
                    ?:  ?=([%"two" %2] t)
                      2
                    ?:  ?=([%"tre" %2] t)
                      3
                    ??  ?=([%"tre" %3] t)
                    3
                  %one
                1


          play:
          lvl   3
          sut   :^  $=  tre=?(%1 %2 %3)
                    tre=$
                    $=  two=?(%1 %2)
                    two=$
                  $=  one=$1
                  one=$
                0|*

          ----
          code  =+  ^=  ott
                    $:  ?(%6647407 %6648436 %7305076)
                        ^-  $
                        ?:  ?=(%"one" +2)
                          one
                        ?:  ?=(%"two" +2)
                          two
                        ??  ?=(%"tre" +2)
                        tre
                    ==
                %+  |=  t=ott
                    ^-  @
                    ?:  ?=([_ %1] t)
                      1
                    ?:  ?=([%"two" %2] t)
                      2
                    ?:  ?=([%"tre" %2] t)
                      3
                    ??  ?=([%"tre" %3] t)
                    3
                  %one
                1


          play:
          lvl   4
          sut   :*  $=  ^=  ott
                        $:  ?(%6647407 %6648436 %7305076)
                            ?:  ?=(%"one" +2)
                              one=$1
                            ?:  ?=(%"two" +2)
                              two=?(%1 %2)
                            tre=?(%1 %2 %3)
                        ==
                    ott=$
                    $=  tre=?(%1 %2 %3)
                    tre=$
                    $=  two=?(%1 %2)
                    two=$
                    $=  one=$1
                    one=$
                    0|*
                ==

          ----
          code  %+  |=  t=ott
                    ^-  @
                    ?:  ?=([_ %1] t)
                      1
                    ?:  ?=([%"two" %2] t)
                      2
                    ?:  ?=([%"tre" %2] t)
                      3
                    ??  ?=([%"tre" %3] t)
                    3
                  %one
                1


          play:
          lvl   4
          sut   :*  $=  ^=  ott
                        $:  ?(%6647407 %6648436 %7305076)
                            ?:  ?=(%"one" +2)
                              one=$1
                            ?:  ?=(%"two" +2)
                              two=?(%1 %2)
                            tre=?(%1 %2 %3)
                        ==
                    ott=$
                    $=  tre=?(%1 %2 %3)
                    tre=$
                    $=  two=?(%1 %2)
                    two=$
                    $=  one=$1
                    one=$
                    0|*
                ==

          ----
          code  |=  t=ott
                ^-  @
                ?:  ?=([_ %1] t)
                  1
                ?:  ?=([%"two" %2] t)
                  2
                ?:  ?=([%"tre" %2] t)
                  3
                ??  ?=([%"tre" %3] t)
                3


          play:
          lvl   5
          sut   :*  ^=  t
                    ^=  ott
                    $:  ?(%6647407 %6648436 %7305076)
                        ?:  ?=(%"one" +2)
                          one=$1
                        ?:  ?=(%"two" +2)
                          two=?(%1 %2)
                        tre=?(%1 %2 %3)
                    ==
                    $=  ^=  ott
                        $:  ?(%6647407 %6648436 %7305076)
                            ?:  ?=(%"one" +2)
                              one=$1
                            ?:  ?=(%"two" +2)
                              two=?(%1 %2)
                            tre=?(%1 %2 %3)
                        ==
                    ott=$
                    $=  tre=?(%1 %2 %3)
                    tre=$
                    $=  two=?(%1 %2)
                    two=$
                    $=  one=$1
                    one=$
                    0|*
                ==



  tire-fish:
  axis  32
  miss  [[%"one" _ _] _]

